ducktape/tests/runner_client.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import os
+import time
+import traceback
+import zmq
+
+from ducktape.tests.message import Request
+from ducktape.tests.loader import TestLoader
+from ducktape.tests.serde import SerDe
+from ducktape.tests.test import TestLogger
+
+from ducktape.tests.result import TestResult, IGNORE, PASS, FAIL
+from ducktape.tests.reporter import SingleResultFileReporter
+from ducktape.utils.local_filesystem_utils import mkdir_p
+
+
+def run_client(port, logger_name, log_dir, debug, max_parallel):
+    client = RunnerClient(port, logger_name, log_dir, debug, max_parallel)
+    client.run()
+
+
+class RunnerClient(object):
+    """Run a single test"""
+
+    def __init__(self, port, logger_name, log_dir, debug, max_parallel):
+        self.serde = SerDe()
+        self.logger = TestLogger(logger_name, log_dir, debug, max_parallel).logger
```

Comment:
Not sure I even understand the second bullet point. Loggers are thread safe anyway, right? The default behavior will just be the first bullet. In any case, I'm confused how this relates to the `max_parallel` parameter?